Source/WebKit/UIProcess/ios/forms/WKFileUploadPanel.mm

```diff
@@ -800,15 +808,17 @@ - (void)repositionContextMenuIfNeeded
         return;
 
     auto inputViewBoundsInWindow = webView->_inputViewBoundsInWindow;
-    if (CGRectIsEmpty(inputViewBoundsInWindow))
+    if (!_isMenuPreviouslyRepositioned && CGRectIsEmpty(inputViewBoundsInWindow))
         return;
 
     // The exact bounds of the context menu container itself isn't exposed through any UIKit API or SPI,
     // and would require traversing the view hierarchy in search of internal UIKit views. For now, just
     // reposition the context menu if its presentation location is covered by the input view.
-    if (!CGRectContainsPoint(inputViewBoundsInWindow, [_view convertPoint:_interactionPoint toView:webView.window]))
+    if (!_isMenuPreviouslyRepositioned && !CGRectContainsPoint(inputViewBoundsInWindow, _interactionPointInWindow))
         return;
 
+    _isMenuPreviouslyRepositioned = !_isMenuPreviouslyRepositioned;
```

Comment:
I don't fully understand why this is being toggled, and not set to a fixed value.

Reply:
It's used when keyboard dismisses and the menu is/was up. The value flips with each function call. So when keyboard shows, it is True at first. When we dismiss it, the menu is still up and has been previously positioned. We only want those early returns if the menu is coming up for the _first time_ or _if it is not up rn_. Otherwise, early returns will trigger and never fix the position to go back down when keyboard is down.

Reply:
What if you passed in a boolean parameter to this method instead that encoded whether the keyboard was being presented or dismissed?

Reply:
I assume you mean smth like a `isKeyboardBeingDismissed` and we pass in the T/F when we call it in `_keyboardDidRequestDismissal` and  `_keyboardDidShow`. I'll try it!